File: scala-3/grocery-todo-list/src/main/scala/com/rbleggi/grocerytodolist/GroceryManager.scala
package com.rbleggi.grocerytodolist

class GroceryManager {
  private var items: List[GroceryItem] = List.empty

  def getItems: List[GroceryItem] = items

  def applyChanges(newItems: List[GroceryItem]): Unit = {
    items = newItems
  }

  def listAll(): Unit = {
    println("\nGrocery List:")
    if (items.isEmpty) {
      println("No items in the list.")
    } else {
      items.foreach(item => println(item))
    }
  }
}

File: java/grocery-todo-list/src/main/java/com/rbleggi/grocerytodolist/GroceryManager.java
package com.rbleggi.grocerytodolist;

import java.util.ArrayList;
import java.util.List;

public class GroceryManager {
    private List<GroceryItem> items = new ArrayList<>();

    public List<GroceryItem> getItems() {
        return new ArrayList<>(items);
    }

    public void applyChanges(List<GroceryItem> newItems) {
        items = new ArrayList<>(newItems);
    }

    public void listAll() {
        System.out.println("\nGrocery List:");
        if (items.isEmpty()) {
            System.out.println("No items in the list.");
        } else {
            items.forEach(System.out::println);
        }
    }
}

